ecs/pod_creator.go

```diff
@@ -27,11 +31,235 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
+
+	options := NewBasicECSPodOptions().
+		SetClient(m.client).
+		SetVault(m.vault).
+		SetStatus(cocoa.Running).
+		SetResources(*resources)
+
+	p, err := NewBasicECSPod(options)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "creating pod")
+	}
+
+	return p, nil
 }
 
 // CreatePodFromExistingDefinition creates a new pod backed by AWS ECS from an
 // existing definition.
 func (m *BasicECSPodCreator) CreatePodFromExistingDefinition(ctx context.Context, def cocoa.ECSTaskDefinition, opts ...*cocoa.ECSPodExecutionOptions) (cocoa.ECSPod, error) {
 	return nil, errors.New("TODO: implement")
 }
+
+// exportTags converts strings into ECS tags.
+func exportTags(tags []string) []*ecs.Tag {
+	ecsTags := []*ecs.Tag{}
+
+	for _, tag := range tags {
+		ecsTag := &ecs.Tag{}
+		ecsTag.SetKey(tag)
+		ecsTags = append(ecsTags, ecsTag)
+	}
+
+	return ecsTags
+}
+
+// exportStrategy converts the strategy and parameter into an ECS placement strategy.
+func exportStrategy(strategy *cocoa.ECSPlacementStrategy, param *cocoa.ECSStrategyParameter) []*ecs.PlacementStrategy {
+	placementStrat := ecs.PlacementStrategy{}
+
+	placementStrat.SetField(utility.FromStringPtr(param))
+
+	if strategy == nil {
+		placementStrat.SetType("")
+	} else {
+		placementStrat.SetType(string(*strategy))
+	}
+
+	return []*ecs.PlacementStrategy{&placementStrat}
+}
+
+// exportEnvVars converts the non-secret environment variables into ECS environment variables.
+func (m *BasicECSPodCreator) exportEnvVars(variables []cocoa.EnvironmentVariable) []*ecs.KeyValuePair {
+	keyValuePairs := []*ecs.KeyValuePair{}
+
+	for _, variable := range variables {
+		if variable.SecretOpts == nil {
+			keyValue := ecs.KeyValuePair{}
+			keyValue.SetName(utility.FromStringPtr(variable.Name)).SetValue(utility.FromStringPtr(variable.Value))
+			keyValuePairs = append(keyValuePairs, &keyValue)
+		}
+	}
+
+	return keyValuePairs
+}
+
+// getSecrets retrieves the secrets from the secret environment variables for the pod.
+func (m *BasicECSPodCreator) getSecrets(merged cocoa.ECSPodCreationOptions) []cocoa.SecretOptions {
+	secrets := []cocoa.SecretOptions{}
+
+	for _, def := range merged.ContainerDefinitions {
+		for _, variable := range def.EnvVars {
+			if variable.SecretOpts != nil {
+				secrets = append(secrets, *variable.SecretOpts)
+			}
+		}
+	}
+
+	return secrets
+}
+
+// createSecrets creates secrets that do not already exist.
+func (m *BasicECSPodCreator) createSecrets(ctx context.Context, secrets []cocoa.SecretOptions) error {
+	if m.vault == nil {
+		return errors.New("vault does not exist")
+	}
+
+	for _, secret := range secrets {
+		if !*secret.Exists {
+			_, err := m.vault.CreateSecret(ctx, secret.PodSecret.NamedSecret)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// translatePodSecrets translates secret options into pod secrets.
+func translatePodSecrets(secrets []cocoa.SecretOptions) []cocoa.PodSecret {
+	podSecrets := []cocoa.PodSecret{}
```

Comment:
You don't need to change it in this PR but as a future reference, using the zero value slice is preferable to initializing an empty slice when all you're doing afterwards is `append`ing values. The reason is that when you do this:
```go
mySlice := []someType{}
```
You initialize memory with an empty slice. _However_, this memory will just be tossed away immediately because the slice has a capacity of 0 and cannot grow anymore. Therefore, `append` will just replace the pointed-to memory with a new block of memory. For example, if you append after initializing it as an empty slice:
```go
mySlice = append(mySlice, <something>
```
This will actually change the `mySlice` pointer to point to a different piece of memory (because the empty slice was initialized with 0 capacity and cannot grow, so it must be replaced with a new block of memory for a new slice), thus wasting the effort of initializing the empty slice in the first place. For that reason, it's preferable to use `var mySlice []someType` when all you're going to do is append some things to it, because it'll be a default nil pointer and won't needlessly create memory for no reason.

I wrote a [tiny example](https://play.golang.org/p/NLE9FzMS7Fd) to show the difference.

Reply:
Understood, I'll keep this in mind for future PRs as well. Thank you!